pkg/prowgen/testdata/zz_fixture_TestGeneratePostSubmitForPromotion_spawn_on_multi01_cluster.yaml

```diff
@@ -0,0 +1,24 @@
+- agent: kubernetes
```

Comment:
This test file proves that the implementation is wrong. The result should generate only one postsubmit.

Reply:
The current implementation of [`generatePostsubmitsForPromotion`](https://github.com/openshift/ci-tools/blob/09cdfa1aefc4a7d11a44e148f362fede533b5689/pkg/prowgen/prowgen.go#L234) creates at least one postsubmit even when neither `multi_arch` nor `additional_architectures` are set. We should either modify it entirely or create a new function like `generatePostsubmitsForPromotionMultiarch` then.

Reply:
We need to clarify the expected output for the following cases before keep modifying the code. What postsubmits and in which cluster do we expect them to be generated for each of the following prowgen configurations?  

1. 
```yaml
multi_arch: true
additional_architectures:
- arm64
```
2.
```yaml
multi_arch: true
```
3.
```yaml
additional_architectures:
- arm64
```
4.
```
{} // nothing at all
```

Reply:
This PR has nothing to do with the `additional_architectures` and the changes are not affecting that feature. Currently, `prowgen` doesn't add any `cluster` field for the normal behavior. This is being done by the sanitize-jobs tool. However, we want to change this behavior when the `multi_arch` value is true.

Reply:
Yeah but a user can still set both `multi_arch` and `additional_architectures`. What postsubmits do we expect for the cases `1.` and `2.` then?
As the current implementation the case `3.` creates 2 postsubmits: one with no cluster defined and another one with `arm01`.
Case `4.` only one without any cluster set.
I agree on leaving `3.` and `4.` untouched but what about `1.` and `2.`?

Reply:
The `additional_architectures` doesn't affect anything. The only thing we need to take care of in this PR is to add the `cluster: multi01` field in the generated postsubmit if the `multi_arch` is true in prowgen's configuration.